crates/wit-component/src/encoding/wit/v2.rs

```diff
@@ -180,52 +226,40 @@ impl InterfaceEncoder<'_> {
     }
 
     fn encode_instance(&mut self, interface: InterfaceId) -> Result<u32> {
-        self.push_instance();
+        encode_instance_contents(self, interface)?;
         let iface = &self.resolve.interfaces[interface];
-        let mut type_order = IndexSet::new();
-        for (_, id) in iface.types.iter() {
-            self.encode_valtype(self.resolve, &Type::Id(*id))?;
-            type_order.insert(*id);
+        let mut instance = self.pop_instance();
+        for (name, nest) in &iface.nested {
+            self.encode_nested(name, nest.id, &mut instance)?;
         }
```

Comment:
Since this loop is duplicated with the below, could this perhaps be inlined into `encode_instance_contents`? (making that a self-recursive method)